andela/ah-django-unchained
# #162948847 Favorite and Unfavorite Articles

#### What does this PR do?
Allows for users to favorite and unfavorite articles
#### Description of Task to be completed?
This endpoint favorites and unfavorites articles depending on the state
`PUT api/articles/<slug:slug>/favorite`
#### How should this be manually tested?

- Clone the repo locally
- cd ah-django-unchained
- git checkout ft-favorite-unfavorite-articles-162948847
- get started with the app as from the README section. Follow the sample .env file supplied to gert further started
- create an article as required.
- once you hit the `PUT api/articles/<slug:slug>/favorite` endpoint, the favorite status should be `True`. this should also apply to the `GET api/articles/:slug` endpoint
<img width="1134" alt="favorite article" src="https://user-images.githubusercontent.com/31338414/51696942-c13f5280-2017-11e9-9c9c-a4e2eec2e37e.png">

- to unfavorite hit the `PUT api/articles/<slug:slug>/favorite` endpoint, and check the same in the `GET api/articles/:slug` endpoint
<img width="1126" alt="favorite article false" src="https://user-images.githubusercontent.com/31338414/51697042-f055c400-2017-11e9-8684-85c00970d9be.png">

#### What are the relevant pivotal tracker stories?
[#162948847](https://www.pivotaltracker.com/story/show/162948847)
